src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java

```diff
@@ -163,6 +164,23 @@ public InnReachResponseDTO cancelItemHold(String trackingId, String centralCode,
     return success();
   }
 
+  @Override
+  public InnReachResponseDTO receiveUnshipped(String trackingId, String centralCode,
+    BaseCircRequestDTO receiveUnshippedRequestDTO) {
+    var transaction = getTransaction(trackingId, centralCode);
+
+    if (transaction.getState() == InnReachTransaction.TransactionState.ITEM_SHIPPED) {
+      throw new IllegalArgumentException("Unexpected transaction state: " + transaction.getState());
+    }
+
+    if (transaction.getState() == InnReachTransaction.TransactionState.ITEM_HOLD) {
+      transaction.setState(RECEIVE_UNANNOUNCED);
+    }
+    transactionRepository.save(transaction);
```

Comment:
lets move `transactionRepository.save(transaction);` to the ITEM_HOLD condition block above, as it's the only place where transaction data is actually being updated

Reply:
Done